src/site/cpp-arrays.rst

```diff
@@ -158,6 +158,107 @@ Arrays vs. Pointers
 Dynamic Arrays
 ==============
 
+* The static arrays require a known size at compile time
+* This is because cpp wants to allocate the RAM on the stack it needs before it starts running
+    * Cpp will manage the memory on the stack for us
+
+* This means determining the size of the array at runtime is problematic
+* The below example will not work
+    * On some platforms this will actually work, but assume it will not
+
+.. code-block:: cpp
+    :linenos:
+
+    int size;
+    std::cin >> size;
+    int foo[size];
+
+
+* However, it is not unreasonable to want to have an array that's size is determined at runtime
+* Fortunately, there is still a way to do this, but we need to allocate the memory on the *heap*
+    * Cpp does not manage this memory for us
+    * We must take care to allocate and deallocate the memory we use in the heap
+
+* In the below example, we can allocate memory in the heap with the ``new`` keyword
+* Notice that the variable ``foo`` is a pointer to an integer
+
+.. code-block:: cpp
+    :linenos:
+
+    int size;
+    std::cin >> size;
+    int* foo = new int[size];
+
+* Even though the variable ``foo`` is an integer pointer, we can still index it like an array as we saw above
+    * ``foo[1]``
+
+* Now we can create arrays that have their size determined at runtime and use them like any other array
+
+.. code-block:: cpp
+    :linenos:
+
+    // Create array
+    int size;
+    std::cin >> size;
+    int* foo = new int[size];
+
+    // Put contents into array
+    for(int i = 0; i < size; i++){
+        foo[i] = i;
+    }
+
+    // print out array contents
+    for(int i = 0; i < size; i++){
+        std::cout << foo[i] << std::endl;
+    }
+
+    // Delete the array
+    delete[] foo;
```

Comment:
Watch it, don't use `delete`. Gotta make sure it's `delete[]`.

I'm so glad the universe invented garbage collection.